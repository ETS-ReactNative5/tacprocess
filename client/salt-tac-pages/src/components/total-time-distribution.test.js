import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { compose } from 'redux';

import Partner from '../util/partner';
import TotalTimeDistribution from '../../src/components/total-time-distribution';

const mountExpect = compose(expect, toJson, mount);

describe('TotalTimeDistribution', () => {
    const AMNH = Partner.partnerByCode('AMNH');
    const RSA = Partner.partnerByCode('RSA');
    const UW = Partner.partnerByCode('UW');

    const timeRequests = {
        AMNH: [12.6, 22.0, 1.9, 25., 58.9, 31, 29.5, 7.8],
        RSA: [0, 0, 0, 78., 0, 99., 0, 0]
    };

    const proposals = timeRequests.AMNH.map((_, i) => (
            {
                timeRequests: [
                    {
                        partner: AMNH,
                        time: 3600 * timeRequests.AMNH[i]
                    },
                    {
                        partner: RSA,
                        time: 3600 * timeRequests.RSA[i]
                    }
                ]
            }
    ));
    proposals[2].timeRequests.push({
                                       partner: UW,
                                       time: 23.9 * 3600
                                   });

    it('should generate the plot for AMNH', () => {
        mountExpect(
                <TotalTimeDistribution height={500}
                                       partner={AMNH}
                                       proposals={proposals}
                                       width={500}/>
        ).toMatchSnapshot();
    });

    it('should generate the plot for RSA', () => {
        mountExpect(
                <TotalTimeDistribution height={500}
                                       partner={RSA}
                                       proposals={proposals}
                                       width={500}/>
        ).toMatchSnapshot();
    });

    it('should generate the plot for UW', () => {
        mountExpect(
                <TotalTimeDistribution height={500}
                                       partner={UW}
                                       proposals={proposals}
                                       width={500}/>
        ).toMatchSnapshot();
    });
});
