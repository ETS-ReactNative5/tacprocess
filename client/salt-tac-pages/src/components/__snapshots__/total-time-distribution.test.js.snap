// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TotalTimeDistribution should generate the plot for AMNH 1`] = `
<TotalTimeDistribution
  height={500}
  partner={
    Partner {
      "code": "AMNH",
      "name": "American Museum of Natural History",
    }
  }
  proposals={
    Array [
      Object {
        "timeRequests": Array [
          Object {
            "partner": Partner {
              "code": "AMNH",
              "name": "American Museum of Natural History",
            },
            "time": 45360,
          },
          Object {
            "partner": Partner {
              "code": "RSA",
              "name": "South Africa",
            },
            "time": 0,
          },
        ],
      },
      Object {
        "timeRequests": Array [
          Object {
            "partner": Partner {
              "code": "AMNH",
              "name": "American Museum of Natural History",
            },
            "time": 79200,
          },
          Object {
            "partner": Partner {
              "code": "RSA",
              "name": "South Africa",
            },
            "time": 0,
          },
        ],
      },
      Object {
        "timeRequests": Array [
          Object {
            "partner": Partner {
              "code": "AMNH",
              "name": "American Museum of Natural History",
            },
            "time": 6840,
          },
          Object {
            "partner": Partner {
              "code": "RSA",
              "name": "South Africa",
            },
            "time": 0,
          },
          Object {
            "partner": Partner {
              "code": "UW",
              "name": "University of Wisconsin",
            },
            "time": 86040,
          },
        ],
      },
      Object {
        "timeRequests": Array [
          Object {
            "partner": Partner {
              "code": "AMNH",
              "name": "American Museum of Natural History",
            },
            "time": 90000,
          },
          Object {
            "partner": Partner {
              "code": "RSA",
              "name": "South Africa",
            },
            "time": 280800,
          },
        ],
      },
      Object {
        "timeRequests": Array [
          Object {
            "partner": Partner {
              "code": "AMNH",
              "name": "American Museum of Natural History",
            },
            "time": 212040,
          },
          Object {
            "partner": Partner {
              "code": "RSA",
              "name": "South Africa",
            },
            "time": 0,
          },
        ],
      },
      Object {
        "timeRequests": Array [
          Object {
            "partner": Partner {
              "code": "AMNH",
              "name": "American Museum of Natural History",
            },
            "time": 111600,
          },
          Object {
            "partner": Partner {
              "code": "RSA",
              "name": "South Africa",
            },
            "time": 356400,
          },
        ],
      },
      Object {
        "timeRequests": Array [
          Object {
            "partner": Partner {
              "code": "AMNH",
              "name": "American Museum of Natural History",
            },
            "time": 106200,
          },
          Object {
            "partner": Partner {
              "code": "RSA",
              "name": "South Africa",
            },
            "time": 0,
          },
        ],
      },
      Object {
        "timeRequests": Array [
          Object {
            "partner": Partner {
              "code": "AMNH",
              "name": "American Museum of Natural History",
            },
            "time": 28080,
          },
          Object {
            "partner": Partner {
              "code": "RSA",
              "name": "South Africa",
            },
            "time": 0,
          },
        ],
      },
    ]
  }
  width={500}
>
  <g
    className="histogram total-time-distribution"
  >
    <Axes
      domain={
        Array [
          0,
          100,
        ]
      }
      height={500}
      margins={
        Object {
          "bottom": 50,
          "left": 50,
          "right": 10,
          "top": 10,
        }
      }
      range={
        Array [
          0,
          3,
        ]
      }
      width={500}
      xAxisTitle="Time (hrs)"
      yAxisTitle="N"
    >
      <g
        className="axes"
      >
        <Axis
          orientation="Top"
          scale={[Function]}
          tickValues={null}
          transform="translate(50, 10)"
        >
          <g
            className="axis"
            transform="translate(50, 10)"
          />
        </Axis>
        <Axis
          orientation="Left"
          scale={[Function]}
          tickValues={null}
          transform="translate(50, 10)"
        >
          <g
            className="axis"
            transform="translate(50, 10)"
          />
        </Axis>
        <Axis
          orientation="Bottom"
          scale={[Function]}
          tickValues={null}
          transform="translate(50, 450)"
        >
          <g
            className="axis"
            transform="translate(50, 450)"
          />
        </Axis>
        <Axis
          orientation="Right"
          scale={[Function]}
          tickValues={null}
          transform="translate(490, 10)"
        >
          <g
            className="axis"
            transform="translate(490, 10)"
          />
        </Axis>
        <Title
          className="x axis title"
          title="Time (hrs)"
          x={270}
          y={490}
        >
          <text
            className="x axis title"
            textAnchor="middle"
            transform="rotate(0, 270, 490)"
            x={270}
            y={490}
          >
            Time (hrs)
          </text>
        </Title>
        <Title
          className="y axis title"
          rotationAngle={-90}
          title="N"
          x={20}
          y={230}
        >
          <text
            className="y axis title"
            textAnchor="middle"
            transform="rotate(-90, 20, 230)"
            x={20}
            y={230}
          >
            N
          </text>
        </Title>
      </g>
    </Axes>
    <rect
      height={293.3333333333333}
      width={94}
      x={50}
      y={156.66666666666669}
    />
    <rect
      height={146.66666666666663}
      width={94}
      x={94}
      y={303.33333333333337}
    />
    <rect
      height={440}
      width={94}
      x={138}
      y={10}
    />
    <rect
      height={146.66666666666663}
      width={94}
      x={182}
      y={303.33333333333337}
    />
    <rect
      height={0}
      width={94}
      x={226}
      y={450}
    />
    <rect
      height={146.66666666666663}
      width={94}
      x={270}
      y={303.33333333333337}
    />
    <rect
      height={0}
      width={94}
      x={314}
      y={450}
    />
    <rect
      height={0}
      width={94}
      x={358}
      y={450}
    />
    <rect
      height={0}
      width={94}
      x={402}
      y={450}
    />
    <rect
      height={0}
      width={94}
      x={446}
      y={450}
    />
  </g>
</TotalTimeDistribution>
`;

exports[`TotalTimeDistribution should generate the plot for RSA 1`] = `
<TotalTimeDistribution
  height={500}
  partner={
    Partner {
      "code": "RSA",
      "name": "South Africa",
    }
  }
  proposals={
    Array [
      Object {
        "timeRequests": Array [
          Object {
            "partner": Partner {
              "code": "AMNH",
              "name": "American Museum of Natural History",
            },
            "time": 45360,
          },
          Object {
            "partner": Partner {
              "code": "RSA",
              "name": "South Africa",
            },
            "time": 0,
          },
        ],
      },
      Object {
        "timeRequests": Array [
          Object {
            "partner": Partner {
              "code": "AMNH",
              "name": "American Museum of Natural History",
            },
            "time": 79200,
          },
          Object {
            "partner": Partner {
              "code": "RSA",
              "name": "South Africa",
            },
            "time": 0,
          },
        ],
      },
      Object {
        "timeRequests": Array [
          Object {
            "partner": Partner {
              "code": "AMNH",
              "name": "American Museum of Natural History",
            },
            "time": 6840,
          },
          Object {
            "partner": Partner {
              "code": "RSA",
              "name": "South Africa",
            },
            "time": 0,
          },
          Object {
            "partner": Partner {
              "code": "UW",
              "name": "University of Wisconsin",
            },
            "time": 86040,
          },
        ],
      },
      Object {
        "timeRequests": Array [
          Object {
            "partner": Partner {
              "code": "AMNH",
              "name": "American Museum of Natural History",
            },
            "time": 90000,
          },
          Object {
            "partner": Partner {
              "code": "RSA",
              "name": "South Africa",
            },
            "time": 280800,
          },
        ],
      },
      Object {
        "timeRequests": Array [
          Object {
            "partner": Partner {
              "code": "AMNH",
              "name": "American Museum of Natural History",
            },
            "time": 212040,
          },
          Object {
            "partner": Partner {
              "code": "RSA",
              "name": "South Africa",
            },
            "time": 0,
          },
        ],
      },
      Object {
        "timeRequests": Array [
          Object {
            "partner": Partner {
              "code": "AMNH",
              "name": "American Museum of Natural History",
            },
            "time": 111600,
          },
          Object {
            "partner": Partner {
              "code": "RSA",
              "name": "South Africa",
            },
            "time": 356400,
          },
        ],
      },
      Object {
        "timeRequests": Array [
          Object {
            "partner": Partner {
              "code": "AMNH",
              "name": "American Museum of Natural History",
            },
            "time": 106200,
          },
          Object {
            "partner": Partner {
              "code": "RSA",
              "name": "South Africa",
            },
            "time": 0,
          },
        ],
      },
      Object {
        "timeRequests": Array [
          Object {
            "partner": Partner {
              "code": "AMNH",
              "name": "American Museum of Natural History",
            },
            "time": 28080,
          },
          Object {
            "partner": Partner {
              "code": "RSA",
              "name": "South Africa",
            },
            "time": 0,
          },
        ],
      },
    ]
  }
  width={500}
>
  <g
    className="histogram total-time-distribution"
  >
    <Axes
      domain={
        Array [
          0,
          100,
        ]
      }
      height={500}
      margins={
        Object {
          "bottom": 50,
          "left": 50,
          "right": 10,
          "top": 10,
        }
      }
      range={
        Array [
          0,
          1,
        ]
      }
      width={500}
      xAxisTitle="Time (hrs)"
      yAxisTitle="N"
    >
      <g
        className="axes"
      >
        <Axis
          orientation="Top"
          scale={[Function]}
          tickValues={null}
          transform="translate(50, 10)"
        >
          <g
            className="axis"
            transform="translate(50, 10)"
          />
        </Axis>
        <Axis
          orientation="Left"
          scale={[Function]}
          tickValues={null}
          transform="translate(50, 10)"
        >
          <g
            className="axis"
            transform="translate(50, 10)"
          />
        </Axis>
        <Axis
          orientation="Bottom"
          scale={[Function]}
          tickValues={null}
          transform="translate(50, 450)"
        >
          <g
            className="axis"
            transform="translate(50, 450)"
          />
        </Axis>
        <Axis
          orientation="Right"
          scale={[Function]}
          tickValues={null}
          transform="translate(490, 10)"
        >
          <g
            className="axis"
            transform="translate(490, 10)"
          />
        </Axis>
        <Title
          className="x axis title"
          title="Time (hrs)"
          x={270}
          y={490}
        >
          <text
            className="x axis title"
            textAnchor="middle"
            transform="rotate(0, 270, 490)"
            x={270}
            y={490}
          >
            Time (hrs)
          </text>
        </Title>
        <Title
          className="y axis title"
          rotationAngle={-90}
          title="N"
          x={20}
          y={230}
        >
          <text
            className="y axis title"
            textAnchor="middle"
            transform="rotate(-90, 20, 230)"
            x={20}
            y={230}
          >
            N
          </text>
        </Title>
      </g>
    </Axes>
    <rect
      height={0}
      width={94}
      x={50}
      y={450}
    />
    <rect
      height={0}
      width={94}
      x={94}
      y={450}
    />
    <rect
      height={0}
      width={94}
      x={138}
      y={450}
    />
    <rect
      height={0}
      width={94}
      x={182}
      y={450}
    />
    <rect
      height={0}
      width={94}
      x={226}
      y={450}
    />
    <rect
      height={0}
      width={94}
      x={270}
      y={450}
    />
    <rect
      height={0}
      width={94}
      x={314}
      y={450}
    />
    <rect
      height={440}
      width={94}
      x={358}
      y={10}
    />
    <rect
      height={0}
      width={94}
      x={402}
      y={450}
    />
    <rect
      height={440}
      width={94}
      x={446}
      y={10}
    />
  </g>
</TotalTimeDistribution>
`;

exports[`TotalTimeDistribution should generate the plot for UW 1`] = `
<TotalTimeDistribution
  height={500}
  partner={
    Partner {
      "code": "UW",
      "name": "University of Wisconsin",
    }
  }
  proposals={
    Array [
      Object {
        "timeRequests": Array [
          Object {
            "partner": Partner {
              "code": "AMNH",
              "name": "American Museum of Natural History",
            },
            "time": 45360,
          },
          Object {
            "partner": Partner {
              "code": "RSA",
              "name": "South Africa",
            },
            "time": 0,
          },
        ],
      },
      Object {
        "timeRequests": Array [
          Object {
            "partner": Partner {
              "code": "AMNH",
              "name": "American Museum of Natural History",
            },
            "time": 79200,
          },
          Object {
            "partner": Partner {
              "code": "RSA",
              "name": "South Africa",
            },
            "time": 0,
          },
        ],
      },
      Object {
        "timeRequests": Array [
          Object {
            "partner": Partner {
              "code": "AMNH",
              "name": "American Museum of Natural History",
            },
            "time": 6840,
          },
          Object {
            "partner": Partner {
              "code": "RSA",
              "name": "South Africa",
            },
            "time": 0,
          },
          Object {
            "partner": Partner {
              "code": "UW",
              "name": "University of Wisconsin",
            },
            "time": 86040,
          },
        ],
      },
      Object {
        "timeRequests": Array [
          Object {
            "partner": Partner {
              "code": "AMNH",
              "name": "American Museum of Natural History",
            },
            "time": 90000,
          },
          Object {
            "partner": Partner {
              "code": "RSA",
              "name": "South Africa",
            },
            "time": 280800,
          },
        ],
      },
      Object {
        "timeRequests": Array [
          Object {
            "partner": Partner {
              "code": "AMNH",
              "name": "American Museum of Natural History",
            },
            "time": 212040,
          },
          Object {
            "partner": Partner {
              "code": "RSA",
              "name": "South Africa",
            },
            "time": 0,
          },
        ],
      },
      Object {
        "timeRequests": Array [
          Object {
            "partner": Partner {
              "code": "AMNH",
              "name": "American Museum of Natural History",
            },
            "time": 111600,
          },
          Object {
            "partner": Partner {
              "code": "RSA",
              "name": "South Africa",
            },
            "time": 356400,
          },
        ],
      },
      Object {
        "timeRequests": Array [
          Object {
            "partner": Partner {
              "code": "AMNH",
              "name": "American Museum of Natural History",
            },
            "time": 106200,
          },
          Object {
            "partner": Partner {
              "code": "RSA",
              "name": "South Africa",
            },
            "time": 0,
          },
        ],
      },
      Object {
        "timeRequests": Array [
          Object {
            "partner": Partner {
              "code": "AMNH",
              "name": "American Museum of Natural History",
            },
            "time": 28080,
          },
          Object {
            "partner": Partner {
              "code": "RSA",
              "name": "South Africa",
            },
            "time": 0,
          },
        ],
      },
    ]
  }
  width={500}
>
  <g
    className="histogram total-time-distribution"
  >
    <Axes
      domain={
        Array [
          0,
          100,
        ]
      }
      height={500}
      margins={
        Object {
          "bottom": 50,
          "left": 50,
          "right": 10,
          "top": 10,
        }
      }
      range={
        Array [
          0,
          1,
        ]
      }
      width={500}
      xAxisTitle="Time (hrs)"
      yAxisTitle="N"
    >
      <g
        className="axes"
      >
        <Axis
          orientation="Top"
          scale={[Function]}
          tickValues={null}
          transform="translate(50, 10)"
        >
          <g
            className="axis"
            transform="translate(50, 10)"
          />
        </Axis>
        <Axis
          orientation="Left"
          scale={[Function]}
          tickValues={null}
          transform="translate(50, 10)"
        >
          <g
            className="axis"
            transform="translate(50, 10)"
          />
        </Axis>
        <Axis
          orientation="Bottom"
          scale={[Function]}
          tickValues={null}
          transform="translate(50, 450)"
        >
          <g
            className="axis"
            transform="translate(50, 450)"
          />
        </Axis>
        <Axis
          orientation="Right"
          scale={[Function]}
          tickValues={null}
          transform="translate(490, 10)"
        >
          <g
            className="axis"
            transform="translate(490, 10)"
          />
        </Axis>
        <Title
          className="x axis title"
          title="Time (hrs)"
          x={270}
          y={490}
        >
          <text
            className="x axis title"
            textAnchor="middle"
            transform="rotate(0, 270, 490)"
            x={270}
            y={490}
          >
            Time (hrs)
          </text>
        </Title>
        <Title
          className="y axis title"
          rotationAngle={-90}
          title="N"
          x={20}
          y={230}
        >
          <text
            className="y axis title"
            textAnchor="middle"
            transform="rotate(-90, 20, 230)"
            x={20}
            y={230}
          >
            N
          </text>
        </Title>
      </g>
    </Axes>
    <rect
      height={0}
      width={94}
      x={50}
      y={450}
    />
    <rect
      height={0}
      width={94}
      x={94}
      y={450}
    />
    <rect
      height={440}
      width={94}
      x={138}
      y={10}
    />
    <rect
      height={0}
      width={94}
      x={182}
      y={450}
    />
    <rect
      height={0}
      width={94}
      x={226}
      y={450}
    />
    <rect
      height={0}
      width={94}
      x={270}
      y={450}
    />
    <rect
      height={0}
      width={94}
      x={314}
      y={450}
    />
    <rect
      height={0}
      width={94}
      x={358}
      y={450}
    />
    <rect
      height={0}
      width={94}
      x={402}
      y={450}
    />
    <rect
      height={0}
      width={94}
      x={446}
      y={450}
    />
  </g>
</TotalTimeDistribution>
`;
